docs/blog/posts/together.md

```diff
@@ -0,0 +1,86 @@
+---
+draft: False
+date: 2024-01-27
+slug: together
+tags:
+  - patching
+  - open source
+authors:
+  - jxnl
+---
+
+# Structured Outputs with Together AI
+
+Open-source LLMS are gaining popularity, and the release of Together's Function calling models, its been easier than ever to get structured outputs.
+
+By the end of this blog post, you will learn how to effectively utilize instructor with Together Ai. But before we proceed, let's first explore the concept of patching.
+
+!!! note "Other Languages"
+
+    This blog post is written in Python, but the concepts are applicable to other languages as well, as we currently have support for [Javascript](
+        https://instructor-ai.github.io/instructor-js) and [Elixir](https://hexdocs.pm/instructor/Instructor.html)
+
+<!-- more -->
+
+## Patching
+
+Instructor's patch enhances a openai api it with the following features:
+
+- `response_model` in `create` calls that returns a pydantic model
+- `max_retries` in `create` calls that retries the call if it fails by using a backoff strategy
+
+!!! note "Learn More"
+
+    To learn more, please refer to the [docs](../../index.md). To understand the benefits of using Pydantic with Instructor, visit the tips and tricks section of the [why use Pydantic](../../why.md) page.
+
+## Together AI
+
+The good news is that Together employs the same OpenAI client, and its models support some of these output modes too!
+
+!!! note "Getting access"
+
+    If you want to try this out for yourself check out the [Together AI](https://www.together.ai/) website. You can get started [here](http://api.together.ai/).
+
+```python
+import os
+import openai
+from pydantic import BaseModel
+import instructor
+
+client = openai.OpenAI(
+    base_url="https://api.together.xyz/v1",
+    api_key=os.environ["TOGETHER_API_KEY"],
+)
+
+
+# By default, the patch function will patch the ChatCompletion.create and ChatCompletion.acreate methods. to support response_model parameter
+client = instructor.patch(client, mode=instructor.Mode.TOOLS)
+
+
+# Now, we can use the response_model parameter using only a base model
+# rather than having to use the OpenAISchema class
+class UserExtract(BaseModel):
+    name: str
+    age: int
+
+
+user: UserExtract = client.chat.completions.create(
+    model="mistralai/Mixtral-8x7B-Instruct-v0.1",
+    response_model=UserExtract,
+    messages=[
+        {"role": "user", "content": "Extract jason is 25 years old"},
```

Comment:
I'm curious, do you need to say "Extract" in the user message? Wouldn't it also work if we said "Jason is 25 years old"?

Reply:
Yes but some weirdness happens with together.

Reply:
Tell me more! (would love to bring this up to the team if there's a bug or something on our end)